packages/messaging/index.js

```diff
@@ -97,10 +97,14 @@ export class Messaging {
      *
      * @param {string} name
      * @param {Record<string, any>} [data]
+     * @param {object} [overrides]
+     * @param {string} [overrides.id] - the ID sent with outgoing requests
      * @return {Promise<any>}
      */
-    request (name, data = {}) {
-        const id = globalThis?.crypto?.randomUUID?.() || name + '.response'
+    request (name, data = {}, options = {}) {
+        // first try the override, then global crypto, falling back to <name>.response
+        const id = options.id || globalThis?.crypto?.randomUUID?.() || name + '.response'
+
```

Comment:
added this 3rd param as an object onto which we can add runtime overrides. Right now just accepting an `id`, but can also be used later to send things like AbortControllers through too.